### PR TITLE
Paginate targets in cache

### PR DESCRIPTION
**Pull request by jefferai**
* Update Go API to support pagination
* Implement pagination of targets in cache

### CODE REVIEW

File: internal/clientcache/internal/daemon/options.go [hunk 1]
```diff
@@ -108,3 +109,10 @@ func WithBoundaryTokenReaderFunc(_ context.Context, fn cache.BoundaryTokenReader
 		return nil
 	}
 }
+
+func WithForceResetSchema(_ context.Context, force bool) Option {
```
Comment (irenarindos) [inline internal/clientcache/internal/daemon/options.go:113]: Can you add doc for this function? And a test?

File: internal/clientcache/internal/db/options.go [hunk 2]
```diff
@@ -67,3 +68,11 @@ func WithDebug(debug bool) Option {
 		return nil
 	}
 }
+
+// WithForceResetSchema provides an optional way to force resetting the cache
```
Comment (irenarindos) [inline internal/clientcache/internal/db/options.go:72]: Can you add a test for this op?

File: internal/clientcache/internal/cache/repository_targets.go [hunk 3]
```diff
@@ -118,44 +129,56 @@ func (r *Repository) refreshTargets(ctx context.Context, u *user, tokens map[Aut
...
-			if numDeleted, err = w.Exec(ctx, "delete from target where id in @ids",
-				[]any{sql.Named("ids", removedIds)}); err != nil {
-				return err
-			}
-		}
-		switch {
-		case unsupportedCacheRequest:
-			if err := upsertRefreshToken(ctx, w, u, resourceType, sentinelNoRefreshToken); err != nil {
-				return err
+	var numUpserted int
```
Comment (irenarindos) [inline internal/clientcache/internal/cache/repository_targets.go:132]: I'm not seeing this set anywhere, but it's later used to write a sys event

File: internal/clientcache/internal/cache/options.go [hunk 4]
```diff
@@ -129,3 +130,12 @@ func WithTestRefreshWaitChs(with *testRefreshWaitChs) Option {
 		return nil
 	}
 }
+
+// WithUseNonPagedListing provides an option for ignoring the resource
+// staleness when performing a search.
+func WithUseNonPagedListing(b bool) Option {
```
Comment (irenarindos) [inline internal/clientcache/internal/cache/options.go:136]: Can you add a test for this?

File: internal/clientcache/internal/daemon/options.go (hunk 1 above)
Comment (jefferai) [reply internal/clientcache/internal/daemon/options.go:113]: Done!

File: internal/clientcache/internal/db/options.go (hunk 2 above)
Comment (jefferai) [reply internal/clientcache/internal/db/options.go:72]: Done!

File: internal/clientcache/internal/cache/repository_targets.go (hunk 3 above)
Comment (jefferai) [reply internal/clientcache/internal/cache/repository_targets.go:132]: Ah thanks, fixed

File: internal/clientcache/internal/cache/options.go (hunk 4 above)
Comment (jefferai) [reply internal/clientcache/internal/cache/options.go:136]: Done!

**Review by irenarindos — APPROVED** [review]

**Review by moduli — COMMENTED** [review]
There are some e2e test failures around listing resources. Are those expected?
```
paginate_credential_library_test.go:119: 
│         	Error Trace:	/src/boundary/testing/internal/e2e/tests/base_with_vault/paginate_credential_library_test.go:119
│         	Error:      	Received unexpected error:
│         	            	exit status 1
│         	Test:       	TestCliPaginateCredentialLibraries
│         	Messages:   	{"context":"Error from controller when performing list on credential libraries","status_code":400,"api_error":{"kind":"InvalidArgument","message":"Improperly formatted identifier.","details":{"request_fields":[{"name":"scope_id","description":"This field must be a valid project scope ID or the list operation must be recursive."}]}}}
```
```
paginate_user_test.go:58: 
│         	Error Trace:	/src/boundary/testing/internal/e2e/tests/base/paginate_user_test.go:58
│         	Error:      	Received unexpected error:
│         	            	exit status 1
│         	Test:       	TestCliPaginateUsers
│         	Messages:   	{"context":"Error from controller when performing list on users","status_code":400,"api_error":{"kind":"InvalidArgument","message":"Improperly formatted identifier.","details":{"request_fields":[{"name":"scope_id","description":"This field must be a valid project scope ID or the list operation must be recursive."}]}}}
```
etc.

**Review by moduli — APPROVED** [review]
e2e tests are fixed

**Pull request merged**